backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -215,15 +215,21 @@ export class ExperimentAssignmentService {
     // experiment level inclusion and exclusion
     const [, exclusionReason] = await this.experimentLevelExclusionInclusion(globalFilteredExperiments, userDoc);
     let monitoredDocument: MonitoredDecisionPoint;
+    let assignmentUnit;
     if (experimentId && experiments.length) {
-      monitoredDocument = await this.monitoredDecisionPointRepository.findOne({
-        where: {
-          site: site,
-          target: target,
-          user: userId,
-        },
-        relations: ['user'],
-      });
+      assignmentUnit = experiments
+      ? experiments.find((exp) => exp.id === experimentId)?.assignmentUnit || experiments[0]?.assignmentUnit
+      : null;
+      if (assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS) {
+        monitoredDocument = await this.monitoredDecisionPointRepository.findOne({
```

Comment:
We are fetching the Monitored decision point only if the experiment exists. This means for all the decision points for which the experiment has not been created we are creating a new monitored decision point document on every mark call.
@ppratikcr7 can you check this scenario